internal/controller/shim_controller.go

```diff
@@ -474,14 +493,15 @@ func (sr *ShimReconciler) handleDeleteShim(ctx context.Context, shim *kwasmv1.Sh
 }
 
 func (sr *ShimReconciler) getNodeListFromShimsNodeSelctor(ctx context.Context, shim *kwasmv1.Shim) (*corev1.NodeList, error) {
+	//TODO (flavio): probably eager optimization - do some pagination?
```

Comment:
I know of clusters with 100s of nodes

Reply:
Even with 5k nodes, it's likely less than 10MB; hence, it's probably not worth it.